### PR TITLE
Feature/map employment dynamics

### PR DESCRIPTION
**Pull request by rantifamilusi**
# Description

Map Employments to dynamics entity
[FAMS3-1563] (https://justice.gov.bc.ca/jira/browse/FAMS3-1563)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested? 

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **FAMS3-1563**

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs [hunk 1]
```diff
@@ -251,7 +357,7 @@ public void PersonSearchCompleted_should_map_to_SSG_SearchApiEvent_correctly()
             Assert.AreEqual(new DateTime(2003, 3, 3), searchEvent.TimeStamp);
             Assert.AreEqual(Keys.EVENT_COMPLETED, searchEvent.EventType);
             Assert.AreEqual(Keys.SEARCH_API_EVENT_NAME, searchEvent.Name);
-            Assert.AreEqual("Auto search processing completed successfully. 2 identifier(s) found.  2 addresses found. 2 phone number(s) found.", searchEvent.Message);
+            Assert.AreEqual("Auto search processing completed successfully. 2 identifier(s) found.  2 addresses found. 2 phone number(s) found. 1 employment(s) found.", searchEvent.Message);
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs]: add name messages here.

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs [hunk 2]
```diff
@@ -84,6 +85,107 @@ public void SSG_SearchApiRequest_should_map_to_PersonSearchRequest_correctly()
...
+        public void Employment_should_map_to_SSG_Employment_correctly()
+        {
+            var employment = new Employment()
+            {
+                Employer = new Employer()
+                {
+                    ContactPerson = "Person",
+                    Name = "Name",
+                    OwnerName = "OwnerName",
+
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs]: Shall we add some Phone data here?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs [hunk 3]
```diff
@@ -88,7 +89,7 @@ public MappingProfile()
                .ForMember(dest => dest.EventType, opt => opt.MapFrom(src => Keys.EVENT_COMPLETED))
                .ForMember(dest => dest.Message,
                           opt => opt.MapFrom(
-                              src => $"Auto search processing completed successfully. {(src.MatchedPerson.Identifiers == null ? 0 : src.MatchedPerson.Identifiers.Count())} identifier(s) found.  {(src.MatchedPerson.Addresses == null ? 0 : src.MatchedPerson.Addresses.Count())} addresses found. {(src.MatchedPerson.Phones == null ? 0 : src.MatchedPerson.Phones.Count())} phone number(s) found."
+                              src => $"Auto search processing completed successfully. {(src.MatchedPerson.Identifiers == null ? 0 : src.MatchedPerson.Identifiers.Count())} identifier(s) found.  {(src.MatchedPerson.Addresses == null ? 0 : src.MatchedPerson.Addresses.Count())} addresses found. {(src.MatchedPerson.Phones == null ? 0 : src.MatchedPerson.Phones.Count())} phone number(s) found. {(src.MatchedPerson.Employments == null ? 0 : src.MatchedPerson.Employments.Count())} employment(s) found."
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs]: add name messages here.

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs (hunk 1 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs]: The PR is for employment data. So did not look at name

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs (hunk 2 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs]: The employment map will not handle phone, we will have to map phone directly .. it's going to a different entity

File: app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs [hunk 4]
```diff
@@ -107,6 +108,26 @@ public MappingProfile()
...
+              .ForMember(dest => dest.AddressLine3, opt => opt.MapFrom(src => (src.Employer == null) ? string.Empty : (src.Employer.Address == null) ? string.Empty : src.Employer.Address.AddressLine3))
+              .ForMember(dest => dest.BusinessName, opt => opt.MapFrom(src => (src.Employer == null) ? string.Empty : src.Employer.Name))
+              .ForMember(dest => dest.CountrySubdivisionText, opt => opt.MapFrom(src => (src.Employer == null) ? string.Empty : (src.Employer.Address == null) ? string.Empty : src.Employer.Address.StateProvince))
+              .ForMember(dest => dest.CountryText, opt => opt.MapFrom(src => (src.Employer == null) ? string.Empty : (src.Employer.Address == null) ? string.Empty : src.Employer.Address.CountryRegion))
+              .ForMember(dest => dest.BusinessOwner, opt => opt.MapFrom(src => (src.Employer == null) ? string.Empty : src.Employer.OwnerName))
+              .ForMember(dest => dest.Notes, opt => opt.MapFrom(src => src.Notes))
+               .ForMember(dest => dest.Website, opt => opt.MapFrom(src => src.Website))
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs]: Maybe it is too late. but i think website should be under employer in Contract. 
thoughts?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs (hunk 3 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs]: The PR is for employment data. So did not look at name

File: app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs (hunk 4 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/MappingProfile.cs]: Yes.. it should be ... it's am employer record

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Employment/SSG_Employment.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Fams3Adapter.Dynamics.Employment
+{
+    public class SSG_Employment :  DynamicsEntity
+    {
+
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Employment/SSG_Employment.cs]: no phone yet?

Comment (rantifamilusi) [reply app/DynamicsAdapter/Fams3Adapter.Dynamics/Employment/SSG_Employment.cs]: We can't add phone entity in this. Phone is a different entity in dynamics. We will handle the mapping with the existing phone entity in dynamics not within the employment entity

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/Mapping/MappingProfileTest.cs]: But we have to test if Employer has an array of Phones, what's the output?

**Pull request merged**